usr/src/lib/libzfs/common/libzfs_sendrecv.c

```diff
@@ -2797,7 +2797,7 @@ zfs_receive_package(libzfs_handle_t *hdl, int fd, const char *destname,
 				*strchr(tofs, '@') = '\0';
 			}
 
-			if (recursive && !flags->dryrun && !flags->nomount) {
+			if (!flags->dryrun && !flags->nomount) {
```

Comment:
The change is not strictly required, of course.  But I'd prefer to have it, because I tautological checks could be confusing / misleading.
I mean
```
if (x) {
...
  if (x) {
...
  }
}
```
Th second condition is redundant and could be misleading.